exporter/elasticsearchexporter/internal/lru/lruset.go

```diff
@@ -0,0 +1,74 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package lru // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/lru"
+
+import (
+	"time"
+
+	"github.com/cespare/xxhash"
+	"github.com/elastic/go-freelru"
+	"go.opentelemetry.io/ebpf-profiler/libpf/xsync"
+)
+
+type void struct{}
+
+var stringHashFn = func(s string) uint32 {
+	return uint32(xxhash.Sum64String(s))
+}
+
+// LockedLRUSet is the interface provided to the LRUSet once a lock has been
+// acquired.
+type LockedLRUSet interface {
+	// CheckAndAdd checks whether the entry is already stored in the cache, and
+	// adds it.
+	// It returns whether the entry should be excluded, as it was already present
+	// in cache.
+	CheckAndAdd(entry string) bool
+}
+
+// LRUSet is an LRU cache implementation that allows acquiring a lock, and
+// checking whether specific keys have already been stored.
+type LRUSet struct {
+	syncMu *xsync.RWMutex[*freelru.LRU[string, void]]
+}
+
+func (l *LRUSet) WithLock(fn func(LockedLRUSet) error) error {
+	if l == nil || l.syncMu == nil {
+		return fn(nilLockedLRUSet{})
+	}
+
+	excluded := l.syncMu.WLock()
```

Comment:
it isn't clear to me why the struct behind the LRU is called `excluded`, especially the struct we passed into `NewRWMutex` in line 54 is called `lru`